### PR TITLE
Update dockerignore

### LINKED ISSUE #372: [Feature Request]: Update .dockerignore
### Feature description

Current image has a lot of unneeded files being added into the image that should be added to the .dockerignore

Here are the current files present in the `/opt/app-root/src` folder:

```
drwxrwx---.   1 default root   25 Jul  7 13:13 .config/
-rw-rw-r--.   1 default root  327 May 16 19:56 .editorconfig
-rw-rw-r--.   1 default root  800 Jul  6 13:20 .env
-rw-rw-r--.   1 default root   83 Jun 28 02:13 .env.development
-rw-rw-r--.   1 default root  854 Jun 15 02:13 .env.local
-rw-rw-r--.   1 default root   42 May 16 19:56 .env.production
-rw-rw-r--.   1 default root   13 May 16 19:56 .env.test
drwxrwxr-x.   1 default root   19 Jul 28 17:34 .git/
drwxrwxr-x.   5 default root   96 May 16 19:56 .github/
-rw-rw-r--.   1 default root  182 Jun 15 02:45 .gitignore
drwxrwxr-x.   2 default root    6 Jul  7 13:13 .npm-global/
-rw-rw-r--.   1 default root   20 May 16 19:56 .npmrc
drwxrwxr-x.   1 default root   19 May  3 16:02 .pki/
-rw-rw-r--.   1 default root  100 May 16 19:56 .prettierrc
drwxrwxr-x.   2 default root   20 May 27 15:25 .vscode/
-rw-rw-r--.   1 default root 1.3K Jul  6 16:24 Dockerfile
-rw-rw-r--.   1 default root  35K Mar  8 18:49 LICENSE
-rw-rw-r--.   1 default root 1.4K Jul  1 15:14 Makefile
-rw-rw-r--.   1 default root 1.6K Jul  1 13:38 README.md
drwxrwxr-x.   1 default root   18 Jul  7 13:15 backend/
drwxrwxr-x.   6 default root  101 May 16 19:56 data/
drwxrwxr-x.   3 default root  168 Jul  6 13:20 docs/
drwxrwxr-x.   4 default root   43 Jun 14 16:35 downloads/
drwxrwxr-x.   7 default root 4.0K Jul  7 13:15 frontend/
drwxrwxr-x.   2 default root   42 Jun 28 02:10 install/
drwxrwxr-x.   4 default root   51 Jun 14 14:01 manifests/
drwxrwxr-x. 388 default root  12K Jul  7 13:15 node_modules/
-rw-rw-r--.   1 default root 179K May 27 13:56 package-lock.json
-rw-rw-r--.   1 default root 2.1K Jul  6 13:20 package.json
drwxrwxr-x.   2 default root   23 May 16 19:56 scripts/
drwxrwxr-x.   3 default root   47 May 16 19:56 tests/
drwxrwxr-x.   4 default root   32 Jun 14 16:35 upload/
```

Here are some things to consider adding to the .dockerignore:

```
.github/
.vscode/
tests/
**/__mocks__/
scripts/
manifests/

**/.git
**/.gitignore

# Unsure if .env files are needed in image
**/.env
**/.env.development
**/.env.production
**/.env.test

.editorconfig
.prettierrc
**/.prettierignore
**/.eslintignore
**/.eslintrc
Makefile
OWNERS
**/README.md
```

### Describe alternatives you've considered

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by lucferbux**
## Description

Follow up of #371 
Fixes #372 
* Add more files to dockerignore 

## How Has This Been Tested?

1. Build a new image
2. Deploy it

## Merge criteria:

- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -2,3 +2,22 @@ frontend/node_modules
 frontend/public
 backend/node_modules
 backend/dist
+
+.github/
+.vscode/
```
[the remark this answers is not in the document]
Comment (lucferbux) [reply .dockerignore:7]: yeah perfect, will update right away.

File: .dockerignore [hunk 2]
```diff
@@ -2,3 +2,22 @@ frontend/node_modules
...
+**/.gitignore
+
+.editorconfig
+.prettierrc
+**/.prettierignore
+**/.eslintignore
+**/.eslintrc
+Makefile
+OWNERS
+**/README.md
```
[the remark this answers is not in the document]
Comment (lucferbux) [reply .dockerignore:23]: yup

**Review by andrewballantyne — COMMENTED** [review]
Still need those local env files ignored -- they may leak info accidentally

**Comment by lucferbux** [comment]
> Still need those local env files ignored -- they may leak info accidentally

I would rather only add the `.env.local` file, since the rest I think are used in the image, right?

**Comment by andrewballantyne** [comment]
> > Still need those local env files ignored -- they may leak info accidentally
> 
> @andrewballantyne I would rather only add the `.env.local` file, since the rest I think are used in the image, right?

Hmmm, I guess that is fair, it would be used to setup the env. It just seems like my local env and your local env will produce different images that have different functionalities, which likely impacts how the image comes out. Maybe that's desired 🤔 I'm not sure if we send any of the data into the env that could be construed as sensitive -- so maybe my previous statement was me worried for nothing.

We can go with what it is and see how it goes I guess.

**Review by andrewballantyne — APPROVED** [review]

**Pull request merged**